dali/operators/random/rng_base.h

```diff
@@ -43,7 +43,9 @@ class RNGBase : public Operator<Backend> {
 
   void RestoreState(const OpCheckpoint &cpt) override {
     if constexpr (std::is_same_v<Backend, CPUBackend>) {
-      rng_ = cpt.CheckpointState<BatchRNG<std::mt19937_64>>();
+      auto snapshot = cpt.CheckpointState<RNGSnapshotCPU64>();
+      DALI_ENFORCE(static_cast<int>(snapshot.rng.size()) == max_batch_size_);
```

Comment:
Could you add here (and in the random crop) a nice a error message about the incorrect checkpoint that is likely due to different batch size between the checkpointed pipeline and the current pipeline?